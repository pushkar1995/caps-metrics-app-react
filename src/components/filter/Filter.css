.filter__container {
    padding: 1rem;
    display: flex;
    justify-content: flex-start;
    align-items: center;
    background-color: #3ea4b4;
}

.filter__select {
    width: 140px;
    font-size: 16px;
    border-radius: 10px;
    padding: 0.3rem;
    box-shadow: 0 0 20px #00000062;
    background-color: #2c93af;
    color: #ebebeb;
}
