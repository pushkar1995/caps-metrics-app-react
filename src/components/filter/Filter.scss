.filter__container {
  padding: 1rem;
  display: flex;
  justify-content: flex-start;
  align-items: center;
  background-color: var(--secondary-color);

  .filter__select {
    width: 140px;
    font-size: 16px;
    border-radius: 10px;
    padding: 0.3rem;
    box-shadow: 0 0 20px #00000062;
    background-color: var(--dark-secondary-color);
    color: var(--white-color);
  }
}
