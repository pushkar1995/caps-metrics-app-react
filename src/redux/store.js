import { configureStore } from '@reduxjs/toolkit';
import countriesReducer from './countriesSlice/countriesSlice';

const store = configureStore({
  reducer: {
    country: countriesReducer,
  },
});

export default store;
