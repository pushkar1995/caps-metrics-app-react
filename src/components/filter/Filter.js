import React, { useState, useEffect, useRef } from 'react';
import { useDispatch } from 'react-redux';
import { setRegion } from '../../redux/countriesSlice';
import worldMap from '../../assets/images/worldMap.png';
import './Filter.css';

const Filter = () => {
  const [filter, setFilter] = useState('');
  const dispatch = useDispatch();
  const selectRef = useRef();

  useEffect(() => {
    if (filter !== '') {
      dispatch(setRegion(filter.toLowerCase()));
    }
  }, [dispatch, filter]);

  const setRefState = () => {
    const newRegion = selectRef.current.value;
    setFilter(newRegion);
  };

  return (
    <div className="filter__container">
      <img src={worldMap} alt="World map" height="150" width="250" />
      <div>
        <select
          name="selected"
          ref={selectRef}
          onChange={setRefState}
          className="filter__select"
        >
          <option defaultValue="Select Region" hidden>
            Select Region
          </option>
          <option value="Asia">Asia</option>
          <option value="Oceania">Oceania </option>
          <option value="Europe"> Europe </option>
          <option value="Americas"> Americas</option>
          <option value="Africa"> Africa</option>
        </select>
      </div>
    </div>
  );
};

export default Filter;
