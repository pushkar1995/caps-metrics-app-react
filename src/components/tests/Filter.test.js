import renderer from 'react-test-renderer';
import { render, screen } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';
import { Provider } from 'react-redux';
import Filter from '../filter/Filter';
import store from '../../redux/store';

describe('Test the Filter', () => {
  it('should match to snapshot', () => {
    const tree = renderer.create(
      <Provider store={store}>
        <Router>
          <Filter />
        </Router>
      </Provider>,
    );
    expect(tree).toMatchSnapshot();
  });
  it('should return the number options', () => {
    render(
      <Provider store={store}>
        <Router>
          <Filter />
        </Router>
      </Provider>,
    );
    const option = screen.getAllByRole('option');
    expect(option).toHaveLength(5);
  });
});
