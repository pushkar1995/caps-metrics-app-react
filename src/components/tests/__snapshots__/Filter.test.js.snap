// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test the Filter should match to snapshot 1`] = `
<div
  className="filter__container"
>
  <img
    alt="World map"
    height="150"
    src="worldMap.png"
    width="250"
  />
  <div>
    <select
      className="filter__select"
      name="selected"
      onChange={[Function]}
    >
      <option
        defaultValue="Select Region"
        hidden={true}
      >
        Select Region
      </option>
      <option
        value="Asia"
      >
        Asia
      </option>
      <option
        value="Oceania"
      >
        Oceania 
      </option>
      <option
        value="Europe"
      >
         Europe 
      </option>
      <option
        value="Americas"
      >
         Americas
      </option>
      <option
        value="Africa"
      >
         Africa
      </option>
    </select>
  </div>
</div>
`;
